examples/ecs/derived_state.rs

```diff
@@ -0,0 +1,209 @@
+//! This example illustrates how to use [`DerivedStates`] for more complex state transitions.
+//!
+//! In this case, we're transitioning from a `Menu` state to an `InGame` state, with the addition
+//! of the capacity to pause and unpause if we are in game using the ESCAPE button on the keyboard.
+
+use bevy::prelude::*;
+
+fn main() {
+    // We start the setup like we did in the states example.
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .init_state::<AppState>()
+        .add_systems(Startup, setup)
+        .add_systems(OnEnter(AppState::Menu), setup_menu)
+        .add_systems(Update, menu.run_if(in_state(AppState::Menu)))
+        .add_systems(OnExit(AppState::Menu), cleanup_menu)
+        // We only want to run the [`setup_game`] function when we enter the [`AppState::InGame`] state, regardless
```

Comment:
It's not clear from this explanation (or the function call) how we control if the state is derived.